rtc/rtc.go

```diff
@@ -81,14 +88,46 @@ func Read(attempts int) error {
 	month := bcdToBin(reg[0x08] & byte(0x7f))
 	year := bcdToBin(reg[0x09] & byte(0x7f))
 
-	timeString := fmt.Sprintf("%02d-%02d-%02dT%02d:%02d:%02d", year, month, day, hours, minutes, seconds)
-	log.Printf("time writing to system clock (in UTC): %s", timeString)
-	cmd := exec.Command("date", "+%y-%m-%dT%H:%M:%S", "--utc", fmt.Sprintf("--set=%s", timeString))
-	out, err := cmd.CombinedOutput()
+	timeString := fmt.Sprintf("20%02d-%02d-%02dT%02d:%02d:%02dZ", year, month, day, hours, minutes, seconds)
+	return time.Parse("2006-01-02T15:04:05Z07:00", timeString)
+}
+
+type RTCState struct {
+	register          [0x14]byte
+	Time              time.Time
+	LowBattery        bool
+	ClockIntegrity    bool
+	BatterySwitchOver bool
```

Comment:
Not a big deal but you could just store registers and calculate the time and flags via flags on demand.